common/network-common/src/main/java/org/apache/spark/network/server/RpcHandler.java

```diff
@@ -38,15 +38,28 @@
    *
    * This method will not be called in parallel for a single TransportClient (i.e., channel).
    *
+   * The rpc *might* included a data stream in <code>streamData</code> (eg. for uploading a large
+   * amount of data which should not be buffered in memory here).  An error while reading data from
+   * the stream ({@link org.apache.spark.network.client.StreamCallback#onData(String, ByteBuffer)})
+   * will fail the entire channel.  A failure in "post-processing" the stream in
+   * {@link org.apache.spark.network.client.StreamCallback#onComplete(String)} will result in an
+   * rpcFailure, but the channel will remain active.
+   *
+   * If streamData is not null, you *must* call <code>streamData.registerStreamCallback</code>
+   * before this method returns.
+   *
    * @param client A channel client which enables the handler to make requests back to the sender
    *               of this RPC. This will always be the exact same object for a particular channel.
    * @param message The serialized bytes of the RPC.
+   * @param streamData StreamData if there is data which is meant to be read via a StreamCallback;
+   *                   otherwise it is null.
    * @param callback Callback which should be invoked exactly once upon success or failure of the
    *                 RPC.
    */
   public abstract void receive(
       TransportClient client,
       ByteBuffer message,
+      StreamData streamData,
```

Comment:
moving discussion from here: https://github.com/apache/spark/pull/21451#discussion_r191628993

@witgo suggested the `message` could be moved inside `streamData` -- any particular reason to do that?  It would work fine to do it that way as well, though I don't see any advantage.  I guess I'm in favor of keeping it this way.